go/vt/vttablet/tabletmanager/rpc_query.go

```diff
@@ -49,6 +50,23 @@ func (tm *TabletManager) ExecuteFetchAsDba(ctx context.Context, query []byte, db
 		_, _ = conn.ExecuteFetch("USE "+sqlescape.EscapeID(dbName), 1, false)
 	}
 
+	// DDL statements may have additional directives/hints embedded in the query
+	allowZeroInDate := false
+	if stmt, err := sqlparser.Parse(string(query)); err == nil {
+		if ddlStmt, ok := stmt.(sqlparser.DDLStatement); ok {
+			if comments := ddlStmt.GetComments(); comments != nil {
+				directives := sqlparser.ExtractCommentDirectives(comments)
+				if _, ok := directives["allowZeroInDate"]; ok {
+					allowZeroInDate = true
+				}
+			}
+		}
+	}
```

Comment:
nit: kindly reduce the cyclomatic complexity

Reply:
@harshit-gangal I've turned it into something else; I don't think the rewrite is any better; the original made more sense to me. Please let me know what you think.

Reply:
I felt it is easier to follow.